sympy/solvers/ode/tests/test_ode.py

```diff
@@ -604,114 +605,115 @@ def test_collect_respecting_exponentials():
 
 
 def test_undetermined_coefficients_match():
-    assert _undetermined_coefficients_match(g(x), x) == {'test': False}
-    assert _undetermined_coefficients_match(sin(2*x + sqrt(5)), x) == \
+    F = lambda eq: NthLinearConstantCoeffUndeterminedCoefficients(SingleODEProblem(eq, f(x), x))._undetermined_coefficients_match(eq,x)
```

Comment:
It's helpful during refactoring if the tests are changed as little as possible so we can be clear that the mostly unchanged tests still pass. This could be:
```
def _undetermined_coefficients_match(eq, x):
    problem = SingleODEProblem(eq, f(x), x)
    solver = NthLinearConstantCoeffUndeterminedCoefficients(problem)
    return solver. _undetermined_coefficients_match(eq, x)
```
Then the rest of the testing code would not need to be changed.